src/libcalamaresui/utils/Paste.cpp

```diff
@@ -40,11 +40,15 @@ logFileContents()
         return QByteArray();
     }
     QFileInfo fi( pasteSourceFile );
-    if ( fi.size() > 16_KiB )
+    sizeLimit = ( sizeLimit < 0 ) ? 1024*1024 : sizeLimit * 1024;            //For KiB to bytes
+    cDebug() << "Log upload size limit was set to " << sizeLimit << " bytes";
+    if ( fi.size() > sizeLimit and sizeLimit > 0 )
     {
-        pasteSourceFile.seek( fi.size() - 16_KiB );
+        // Fixme : this following line is not getting pasted
+        cDebug() << "Only last " << sizeLimit << " bytes of log file (" << fi.size() << ") uploaded" ;
```

Comment:
For debugging this (and the FIXME) you should write a static, C-style function that does a `stat()` on the log file (get its name via `Logger::logFile()`) and `fprintf(stderr, ..)` the results.

What I suspect is this: `QFileInfo` does the `stat()` call in its constructor; let's say it finds the file to be exactly 70000 bytes long. Now you log one extra line, so you have about 70060 bytes in the file. Below, we `seek()` to the size from the fileinfo, minus the size limit: but that is `70000 - sizeLimit`, not the **current** size of the file after that debug line! So you never see that line in the uploaded log, even though it might be there in the log at the moment of upload.

Another way of testing this is to seek to `fi.size()-sizeLimit+10`, or to read `sizeLimit+10`, just so see if the bytes are there. Come to think of it, reading `sizeLimit+1_KiB` might even be a good thing **anyway** in case new logging was done from another thread.